src/math/vec2.cpp

```diff
@@ -0,0 +1,102 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#include "vec2.h"
+
+namespace Bald::Math {
+
+    Vec2 Vec2::operator+(const Vec2 &other) const noexcept {
+        return Vec2(m_x + other.m_x, m_y + other.m_y);
+    }
+
+
+    Vec2 Vec2::operator-(const Vec2 &other) const noexcept {
+        return Vec2(m_x - other.m_x, m_y - other.m_y);
+    }
+
+
+    Vec2 Vec2::operator*(float multiplier) const noexcept {
+        return Vec2(m_x * multiplier, m_y * multiplier);
+    }
+
+
+    Vec2 &Vec2::operator+=(const Vec2 &other) noexcept {
+        m_x += other.m_x;
+        m_y += other.m_y;
+
+        return *this;
+    }
+
+
+    Vec2 &Vec2::operator-=(const Vec2 &other) noexcept {
+        *this += other.Reverse();
+
+        return *this;
+    }
+
+
+    Vec2 &Vec2::operator*=(float multiplier) noexcept {
+        m_x *= multiplier;
+        m_y *= multiplier;
+
+        return *this;
+    }
+
+
+    bool Vec2::operator==(const Vec2 &other) const noexcept {
+        return m_x == other.m_x && m_y == other.m_y;
+    }
+
+
+    bool Vec2::operator!=(const Vec2 &other) const noexcept {
+        return !(m_x == other.m_x && m_y == other.m_y);
+    }
+
+
+    Vec2 Vec2::MakeUnitVec(const Vec2 &vec) noexcept {
+        return Vec2(vec.m_x / vec.Len(), vec.m_y / vec.Len());
+    }
+
+
+    Vec2 &Vec2::Normalize() noexcept {
+        float len = Len();
+        if (len != 0.0f) {
+            m_x /= len;
+            m_y /= len;
+        }
+        return *this;
+    }
+
+
+    float Vec2::Len() const noexcept {
+        return static_cast<float>(sqrt(pow(m_x, 2.0f) + pow(m_y, 2.0f)));
```

Comment:
function pow is very slow it this case simple multiplication will be better

Reply:
done